src/docker.ts

```diff
@@ -36,6 +53,21 @@ const saveDockerImages = async (): Promise<void> => {
       `Cache miss occurred on the primary key ${key}. Not saving cache as ` +
         "read-only option was selected."
     );
+    /* Access the private method, getCacheEntry, internal to the GitHub Actions
+     * cache action. Use it to check if a cache with our key has been saved between
+     * when we checked in loadDockerImages and now.
+     */
+  } else if (
+    await (<CacheWithGetCacheEntry>cache)["getCacheEntry"](
+      [key],
+      [DOCKER_IMAGES_PATH]
+    )
```

Comment:
We could instead test whether [`key === await restoreCache([""], key, [], { lookupOnly: true })`](https://github.com/actions/toolkit/blob/%40actions/cache%403.2.2/packages/cache/src/cache.ts#L56-L72) since [`@actions/cache` added support for `lookupOnly`](https://github.com/actions/toolkit/blob/%40actions/cache%403.2.2/packages/cache/src/options.ts#L63-L70) in [v3.2.0](https://github.com/actions/toolkit/blob/%40actions/cache%403.2.2/packages/cache/RELEASES.md#320). That would let us use the public API of `@actions/cache` rather than jump through hoops to call an internal method and allow us to undo almost all of the other changes.